### PR TITLE
test(tabs): add accessibility tests

### PR DESCRIPTION
**Pull request by ZhuoyuJin**
### Proposed behaviour

- Add `accessibility` Cypress tests for the `Tabs` component to use the new cypress-component-react framework for testing.

### Current behaviour
Old approach for accessibility tests

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added test.file
- [x] Check if the `tabs.test.js` file passed
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed
- [x] Run `npm run build-storybook` -> `npx sb extract` -> `npx http-server storybook-static -p 9001` and run `npx cypress run --e2e` and check that `Tabs` tests are not running twice.

### CODE REVIEW

File: src/components/tabs/tabs-test.stories.tsx [hunk 1]
```diff
@@ -110,3 +111,275 @@ export const Default = (args: TabsProps) => {
...
+            onChange={() => setInfos({ ...infos, three: !infos.three })}
+            checked={infos.three}
+          />
+        </Tab>
+      </Tabs>
+    </div>
+  );
+};
+
+// eslint-disable-next-line react/prop-types
```
Comment (DlgSHi) [inline src/components/tabs/tabs-test.stories.tsx:287]: why we need to disable prop-types?

File: src/components/tabs/tabs-test.stories.tsx [hunk 2]
```diff
@@ -110,3 +111,275 @@ export const Default = (args: TabsProps) => {
 };
 
 Default.storyName = "default";
+
+export const TabsComponent = ({ ...props }) => {
```
Comment (DlgSHi) [inline src/components/tabs/tabs-test.stories.tsx:115]: exclude all `test` components from showing in storybook

File: src/components/tabs/tabs.test.js [hunk 3]
```diff
@@ -617,26 +317,6 @@ context("Testing Tabs component", () => {
...
-        tabById(1)
-          .trigger("mouseover")
-          .then(() => {
-            tooltipPreview().should(
-              "have.text",
-              `• ${validationMessage}\n• ${validationMessage}\n• ${validationMessage}`
-            );
-          });
-      }
-    );
```
Comment (stephenogorman) [inline src/components/tabs/tabs.test.js:638]: Why is this test removed?

Comment (ZhuoyuJin) [reply src/components/tabs/tabs.test.js:638]: Validation prop has been removed, no such text format can be found in the master storybook for Tabs component

File: src/components/tabs/tabs.test.js [hunk 4]
```diff
@@ -835,4 +515,162 @@ context("Testing Tabs component", () => {
...
+    });
+
+    it("should pass accessibility tests for Tabs WithStringValidationsSummarised story", () => {
+      CypressMountWithProviders(<stories.WithStringValidationsSummarised />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for Tabs Responsive story", () => {
+      CypressMountWithProviders(<stories.Responsive />);
```
Comment (DlgSHi) [inline src/components/tabs/tabs.test.js:671]: shall we provide some viewport for this test?

**Review by DlgSHi — APPROVED** [review]

File: src/components/tabs/tabs-test.stories.tsx [hunk 5]
```diff
@@ -110,3 +112,274 @@ export const Default = (args: TabsProps) => {
...
+          key="tab-5"
+        >
+          Content for tab 5
+        </Tab>
+      </Tabs>
+    </div>
+  );
+};
+
+export const TabsComponentValidations = ({ ...props }) => {
```
Comment (Parsium) [inline src/components/tabs/tabs-test.stories.tsx:180]: Likewise here too

```suggestion
export const TabsComponentValidations = ({ ...props }: TabsProps) => {
```

**Review by Parsium — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Pull request merged**